app/secrets.py

```diff
@@ -15,8 +13,44 @@
     "EQ_SERVER_SIDE_STORAGE_DATABASE_PASSWORD",
 ]
 
+PURPOSE_AUTHENTICATION = 'authentication'
+PURPOSE_SUBMISSION = 'submission'
+
 
 def validate_required_secrets(secrets):
     for required_secret in EXPECTED_SECRETS:
-        if required_secret not in secrets:
+        if required_secret not in secrets['secrets']:
             raise Exception("Missing Secret [{}]".format(required_secret))
+
+
+class SecretStore:
+    def __init__(self, secrets):
+        if 'secrets' in secrets:
+            self.secrets = secrets['secrets']
```

Comment:
maybe the `if` statements should be removed and replace with `self.secrets = secrets['secrets'] or None` so at least the attribute is always set at `__init__` even if to `None` ?